src/tasks/index.ts

```diff
@@ -0,0 +1,174 @@
+import type { Schema } from 'ajv';
+import * as cronParser from 'cron-parser';
+import { tasks as tasksEnv } from '../config-loader/env';
+import type * as Db from '../database-layer/db';
+import { BadRequestError } from '../sbvr-api/errors';
+import { addPureHook } from '../sbvr-api/hooks';
+import { PinejsClient } from '../sbvr-api/sbvr-utils';
+import type * as sbvrUtils from '../sbvr-api/sbvr-utils';
+import { ajv, apiRoot, channel } from './common';
+import type { TaskHandler } from './types';
+import { Worker } from './worker';
+
+export * from './types';
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const modelText: string = require('./model.sbvr');
+
+export const config = {
+	models: [
+		{
+			modelName: apiRoot,
+			apiRoot,
+			modelText,
+			customServerCode: exports,
+		},
+	] as sbvrUtils.ExecutableModel[],
+};
+
+// Poll every second and also execute on new task insert
+// Only poll or execute on triggers if worker is not already at max concurrency
+async function createTrigger(tx: Db.Tx): Promise<void> {
+	await tx.executeSql(`
+		CREATE OR REPLACE FUNCTION notify_task_insert()
+		RETURNS TRIGGER AS $$
+		BEGIN
+			PERFORM pg_notify('${channel}', NEW.id::text);
+			RETURN NEW;
+		END;
+		$$ LANGUAGE plpgsql;
+	`);
+
+	// Only trigger if task is queued and not scheduled
+	await tx.executeSql(`
+		CREATE OR REPLACE TRIGGER task_insert_trigger
+		AFTER INSERT ON task
+		FOR EACH ROW WHEN (NEW.status = 'queued' AND NEW."is scheduled to execute on-time" IS NULL)
+		EXECUTE FUNCTION notify_task_insert();
+	`);
+}
+
+// Partial index for polling
+async function createIndexes(tx: Db.Tx): Promise<void> {
+	await tx.executeSql(`
+		CREATE INDEX IF NOT EXISTS idx_task_poll ON task USING btree (
+			"is executed by-handler",
+			"is scheduled to execute on-time" ASC,
+			"priority" DESC,
+			"id" ASC
+		) WHERE status = 'queued';
+	`);
+}
+
+let worker: Worker | null = null;
+export async function setup(db: Db.Database, tx: Db.Tx): Promise<void> {
+	// Async task functionality is only supported on Postgres
+	if (db.engine !== 'postgres') {
+		return;
+	}
+
+	// Create trigger function if it doesn't exist
+	await createTrigger(tx);
+
+	// Create indexes if they don't exist
+	await createIndexes(tx);
```

Comment:
This should be able to be done by defining and `initSql` in the model definition the same way as user models would